Remotion/Core/Core/Reflection/TypeDiscovery/BaseTypeCache.cs

```diff
@@ -69,13 +70,19 @@ public static BaseTypeCache Create (IEnumerable<Type> types)
             } while (nextBaseType != null);
           }
 
-          foreach (var interfaceType in type.GetInterfaces())
+          var implementedInterfaces = type.GetInterfaces();
+          if (implementedInterfaces.Length > 0)
           {
-            var normalizedInterfaceType = GetNormalizedType(interfaceType);
-            if (enqueuedTypes.Add(normalizedInterfaceType))
-              typesToProcess.Enqueue(normalizedInterfaceType);
+            processedInterfaces.Clear();
+            foreach (var implementedInterface in implementedInterfaces)
+            {
+              var normalizedInterfaceType = GetNormalizedType(implementedInterface);
+              if (enqueuedTypes.Add(normalizedInterfaceType))
+                typesToProcess.Enqueue(normalizedInterfaceType);
 
-            interfaces.Add(new KeyValuePair<Type, Type>(normalizedInterfaceType, type));
+              if (processedInterfaces.Add(normalizedInterfaceType))
```

Comment:
What did you learn about the performance characteristicts for small Hashsets?

Reply:
Too much variability to make a decisive decision so I choose the simpler one with the hash set.